src/init.cpp

```diff
@@ -1668,6 +1668,7 @@ bool AppInit2(boost::thread_group &threadGroup, CScheduler &scheduler) {
                 delete pcoinsdbview;
                 delete pcoinscatcher;
                 delete pblocktree;
+                sigma::CSigmaState::GetState()->Reset();
```

Comment:
Wold it be more logical to move this to UnloadBlockIndex() function?

Reply:
There is no logical evidence that moving this into UnloadBlockIndex will make the program any better as UnloadBlockIndex is called exactly once 4 lines above. 
It maybe more aesthetic to put it there. That's why I am doing so.

Reply:
Saying logical, I mean all such calls are at UnloadBlockIndex.